lib/internal/child_process.js

```diff
@@ -516,6 +516,12 @@ function setupChannel(target, channel) {
     // a message.
     target._send({ cmd: 'NODE_HANDLE_ACK' }, null, true);
 
+    // It can happen the handle is not received because of some error on
+    // ancillary data reception such as MSG_CTRUNC. In this case don't try to
+    // perform any conversion.
+    if (!handle)
```

Comment:
hmm... I understand the reasoning behind this but I'm wondering if this is the right approach. If the handle is invalid at this point, should we treat it as an error condition?